simespresso/io_QE/espresso_class.py

```diff
@@ -0,0 +1,787 @@
+                            if z_count == z_points:
+                                break
+
+                    line = file_iter.next()
+
+        except StopIteration:
+            print('EOF')
+        if x_count<x_points or y_count<y_points or z_count<z_points:
+            logging.debug('Got fewer points than expected:read {0} of {1} x, {2} of {3} y, {4} of {5} z'.format(x_count,x_points,y_count,y_points,z_count,z_points))
+        return charge_density
+
+
+    def WriteEspressoInputFile(self,file_name):
+        """
+        :param file_name: name of the input file to write
+        :return:
+        """
+        SP = self.SP
+        pc = self.pc
+        #write parameters for a particular working input file
+
+        print('attempting to write '+file_name)
+        with open(file_name, 'w') as f:
+            try:
+                #CONTROL section
+                #apparently a comma is not required at the end of every line
+                line = '&CONTROL\n'  #calculation
+                f.write(line)
+                if CUBA.TORQUE in SP:
+                    line = '\t calculation=\''+str(SP[CUBA.TORQUE])+'\'\n'   #calculation
+                    f.write(line)
+                if CUBA.ZETA_POTENTIAL in SP:
+                    line = '\t restart_mode=\''+str(SP[CUBA.ZETA_POTENTIAL])+'\'\n' #restart_mode
+                    f.write(line)
+                if CUBA.YOUNG_MODULUS in SP:
+                    line = '\t pseudo_dir=\''+str(SP[CUBA.YOUNG_MODULUS])+'\'\n'  #pseudo dir
+                    f.write(line)
+                if CUBA.VOLUME_FRACTION in SP:
+                    line = '\t prefix=\''+str(SP[CUBA.VOLUME_FRACTION])+'\'\n'  #prefix
+                    f.write(line)
+                if CUBA.AMPHIPHILICITY in SP:
+                    line = '\t tprnfor='+str(SP[CUBA.AMPHIPHILICITY])+'\n'  #tprnfor
+                    f.write(line)
+                if CUBA.NUMBER_OF_TIME_STEPS in SP:
+                    line = '\t max_seconds='+str(int(SP[CUBA.NUMBER_OF_TIME_STEPS]))+'\n'
+                    f.write(line)
+                if CUBA.DIRECTION in SP:
+                    line = '\t outdir=\''+str(SP[CUBA.DIRECTION])+'\'\n'  #outdir
+                    f.write(line)
+                line = '/\n'
+                f.write(line)
+
+                #SYSTEM section
+                line = '&SYSTEM\n'  #calculation
+                f.write(line)
+                if CUBA.ROLLING_FRICTION in SP:
+                    line = '\t ibrav='+str(SP[CUBA.ROLLING_FRICTION])+'\n'  #outdir
+                    f.write(line)
+                if CUBA.ORIGINAL_POSITION in SP:
+                    line = '\t celldm(1)='+str(SP[CUBA.ORIGINAL_POSITION][0])+'\n'
+                    f.write(line)
+                    line = '\t celldm(2)='+str(SP[CUBA.ORIGINAL_POSITION][1])+'\n'
+                    f.write(line)
+                    line = '\t celldm(3)='+str(SP[CUBA.ORIGINAL_POSITION][2])+'\n'
+                    f.write(line)
```

Comment:
see here: https://github.com/simphony/simphony-espresso-dft/pull/7/files?diff=split#r42113584
We need to use the CUBA extensions, or even better start pushing new CUBA into SimPhoNy common (with pull requests!)